### PR TITLE
SEQNG-911: Subsystem button colors

### PR DESCRIPTION
**Pull request by sraaphorst**
This PR changes the color of the subsystem buttons in an observation step, using:

- Blue: Idle / not configured.
- Yellow: Configuring.
- Green: Configured.

This is done by introducing a new state for subsystem configuration, namely `ResourceRunComplete`.

One thing I'm not entirely happy about is that when the table is first loaded, the map of resources to steps is an empty map; after that point, it becomes populated, and resources are moved to `ResourceRunIdle`.

One thought, given that I don't think we want to assume what resources are going to be used in advance: we use `Map.empty.withDefaultValue(ResourceRunOperation.ResourceRunIdle)` for the resource state map in `TabOperations.Default`. The issue with this is that the return from `SortedMap.withDefaultValue` is a `Map` instead of a `SortedMap` (although it will still be sorted: the underlying implementation remains a `SortedMap`).

I'm not entirely sure why we are using a `SortedMap` here: does it give us a significant advantage? In any case, the `SortedMap` is spread across a number of files (but as far as I can tell, not required), I'm not going to touch this until it's discussed further.

I'm including a video showing the change to this PR.

seqng911

### CODE REVIEW

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/OperationsStateHandler.scala [hunk 1]
```diff
@@ -75,7 +75,12 @@ class OperationsStateHandler[M](modelRW: ModelRW[M, SequencesOnDisplay])
                                .set(StartFromOperation.StartFromInFlight)))
 
     case RunResourceComplete(id, _, r) =>
-      updated(value.markOperations(id, TabOperations.resourceRun(r).set(none)))
+      updated(
+        value.markOperations(
+          id,
+          TabOperations
+            .resourceRun(r).
```
Comment (cquiroz) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/OperationsStateHandler.scala]: looks odd to have a dot `.` at both ends.

**Comment by cquiroz** [comment]
Looks great. Could we add a `check mark` along the green color?

**Comment by sraaphorst** [comment]
@cquiroz Sure. That would probably be clearer. I'll go ahead and give that a try.

**Comment by sraaphorst** [comment]
@cquiroz Here is a video with the checkmarks:
checkmarks

**Pull request merged**